src/flask_cognito_lib/services/cognito_svc.py

```diff
@@ -96,6 +97,116 @@ def exchange_code_for_token(
             "code_verifier": code_verifier,
         }
 
+        return self._request_token(data)
+
+    def refresh_token(
+        self,
+        refresh_token: str,
+    ) -> CognitoTokenResponse:
+        """
+        Exchange a refresh token for a new set of tokens
+
+        Parameters:
+        -----------
+        refresh_token : str
+            The refresh token to exchange for a new set of tokens
+
+        Returns:
+        --------
+        CognitoTokenResponse
+            A dataclass that holds the token response from Cognito
+
+        Raises:
+        -------
+        CognitoError
+            If the request to the endpoint fails
+            If the endpoint returns an error code
+        """
+        data = {
+            "grant_type": "refresh_token",
+            "client_id": self.cfg.user_pool_client_id,
+            "refresh_token": refresh_token,
+        }
+
+        return self._request_token(data)
+
+    def revoke_refresh_token(
+        self,
+        refresh_token: str,
+    ) -> None:
+        """
+        Revoke a refresh token
+
+        Parameters:
+        -----------
+        refresh_token : str
+            The refresh token to revoke
+
+        Raises:
+        -------
+        @see https://docs.aws.amazon.com/cognito/latest/developerguide/revocation-endpoint.html#revoke-sample-response
+
+        CognitoError
+            If the token isn't present in the request or if the feature is disabled for the app client
+            If the token that Amazon Cognito sent in the revocation request isn't a refresh token
+            If the client credentials aren't valid
+        """  # noqa
+        data = {
+            "token": refresh_token,
+        }
+
+        self._request(url=self.cfg.revoke_endpoint, data=data)
+
+    def _request_token(self, data: dict) -> CognitoTokenResponse:
+        """
+        Request a token from the Cognito token endpoint
+
+        Parameters
+        ----------
+        data : dict
+            The data to be sent as part of the request
+
+        Returns
+        -------
+        CognitoTokenResponse
+            A dataclass that holds the token response from Cognito
+
+        Raises:
+        -------
+        CognitoError
+            If the request to the endpoint fails
+            If the endpoint returns an error code
+        """
+        response = self._request(url=self.cfg.token_endpoint, data=data)
+        response_json = response.json()
+
+        if "error" in response_json:
+            raise CognitoError(f"Cognito error : {response_json['error']}")
+
+        return CognitoTokenResponse(**response_json)
```

Comment:
According to the [Cognito docs](https://docs.aws.amazon.com/cognito/latest/developerguide/revocation-endpoint.html#revoke-sample-response), it seems like it's possible that a revoke request can include `error_description` in the response. Should we update the `CognitoTokenResponse` adding this as an optional field?

> A successful response contains an empty body. The error response is a JSON object with an error field and, in some cases, an error_description field.

Reply:
Good catch, will add a graceful handling for this case too.

Reply:
Added handling